src/charm.py

```diff
@@ -160,22 +157,20 @@ def _configure_kubeproxy(self, event):
             external_cloud_provider=self.external_cloud_provider,
         )
 
-    def _configure_labels(self):
-        """Configure labels."""
-        if not os.path.exists("/root/.kube/config"):
-            log.info("Waiting for kubeconfig before configuring labels")
-            return
-
-        status.add(ops.MaintenanceStatus("Configuring node labels"))
-
+    def _apply_node_labels(self):
+        """Apply node labels."""
+        status.add(ops.MaintenanceStatus("Apply Node Labels"))
+        node = self.get_node_name()
         if self.label_maker.active_labels() is not None:
             self.label_maker.apply_node_labels()
+            log.info("Node %s labelled successfully", node)
+        else:
+            log.info("Node %s not yet labelled", node)
```

Comment:
Should we also return an error or do something that re-triggers the reconciliation in the future? This seems like an error to me because if we receive `None` here it either means `NodeLabelError` or `JSONDecodeError`.

Reply:
well, you're absolutely right...  I guess we should raise an error and post  `WaitingStatus()`